File: src/day1/main.cpp
#include"ReadFile.h"
#include<iostream>
#include<vector>
#include<algorithm>

using std::cout;
using std::endl;

void readInput(std::vector<int> *v)
{
	mhr::ReadFile inf("input.txt");
	std::string ll;
	while(inf.readLine(ll))
	{
		if(ll.length() > 0)
		{
			int ii = std::stoi(ll);
			v->push_back(ii);
		}
		else
			v->push_back(-1);
	}
	inf.close();
	v->push_back(-1);
}

void part1()
{
	std::vector<int> ll;
	readInput(&ll);
	int cnt = ll.size() - 1;

	int maxcal = 0;
	int cal = 0;
	for(int i = 0; i < cnt; i++)
	{
		if(ll[i] >= 0)
			cal += ll[i];
		else
		{
			if(cal > maxcal) maxcal = cal;
			cal = 0;
		}
	}
	cout << "PART 1 maxcal = " << maxcal << endl;
}

void part2()
{
	std::vector<int> ll;
	readInput(&ll);
	int cnt = ll.size() - 1;

	std::vector<int> totals;

	int cal = 0;
	for(int i = 0; i < cnt; i++)
	{
		if(ll[i] >= 0)
			cal += ll[i];
		else
		{
			totals.push_back(cal);
			cal = 0;
		}
	}
	std::sort(totals.begin(), totals.end());
	int total = totals[totals.size() - 3] + totals[totals.size() - 2] + totals[totals.size() - 1];
	cout << "PART 2 total = " << total << endl;
}

int main()
{
	part1();
	part2();
	return 0;
}

File: src/day1/Main.kt
import java.io.File
import java.util.*

fun readAllLines(fileName: String): List<String> = File(fileName).bufferedReader().readLines()

fun part1()
{
    //val fileName = "D:\\progs\\advent\\2022\\day1\\sample_input.txt"
    val fileName = "D:\\progs\\advent\\2022\\day1\\input.txt"
    val ll = readAllLines(fileName)

    var maxCal = 0
    var cal = 0
    for(ss in ll)
    {
        if(ss.isNotEmpty())
        {
            cal += ss.toInt()
        }
        else
        {
            if(cal > maxCal) maxCal = cal
            cal = 0
        }
    }
    println("PART 1 maxCal = $maxCal")
}

fun part2()
{
    //val fileName = "D:\\progs\\advent\\2022\\day1\\sample_input.txt"
    val fileName = "D:\\progs\\advent\\2022\\day1\\input.txt"
    val ll = readAllLines(fileName)

    var totals =  Vector<Int>();
    var cal = 0
    for(ss in ll)
    {
        if(ss.isNotEmpty())
        {
            cal += ss.toInt()
        }
        else
        {
            totals.addElement(cal)
            cal = 0
        }
    }
    totals.sort()
    val total = totals[totals.size - 3] + totals[totals.size - 2] + totals[totals.size - 1];
    println("PART 2 total = $total")
}

fun main()
{
    part1()
    part2()
}
